programs/drift/src/controller/insurance.rs

```diff
@@ -474,6 +477,89 @@ pub fn admin_remove_insurance_fund_stake(
     Ok(withdraw_amount)
 }
 
+pub fn admin_transfer_insurance_fund_stake(
+    insurance_vault_amount: u64,
+    n_shares: u128,
+    target_insurance_fund_stake: &mut InsuranceFundStake,
+    user_stats: &mut UserStats,
+    spot_market: &mut SpotMarket,
+    now: i64,
+    admin_pubkey: Pubkey,
+) -> DriftResult<u64> {
+    apply_rebase_to_insurance_fund(insurance_vault_amount, spot_market)?;
+
+    let total_if_shares_before = spot_market.insurance_fund.total_shares;
+    let user_if_shares_before = spot_market.insurance_fund.user_shares;
+
+    let if_shares_before = total_if_shares_before.safe_sub(user_if_shares_before)?;
+    let target_if_shares_before = target_insurance_fund_stake.checked_if_shares(spot_market)?;
+    validate!(
+        if_shares_before >= n_shares,
+        ErrorCode::InsufficientIFShares,
+        "if_shares_before={} < n_shares={}",
+        if_shares_before,
+        n_shares
+    )?;
+
+    spot_market.insurance_fund.user_shares =
+        spot_market.insurance_fund.user_shares.safe_add(n_shares)?;
+
+    target_insurance_fund_stake.increase_if_shares(n_shares, spot_market)?;
+
+    let target_if_shares_after = target_insurance_fund_stake.checked_if_shares(spot_market)?;
+
+    if spot_market.market_index == 0 {
+        user_stats.if_staked_quote_asset_amount = if_shares_to_vault_amount(
+            target_if_shares_after,
+            spot_market.insurance_fund.total_shares,
+            insurance_vault_amount,
+        )?;
+    }
+
+    let withdraw_amount = if_shares_to_vault_amount(
+        n_shares,
+        spot_market.insurance_fund.total_shares,
+        insurance_vault_amount,
+    )?;
+
+    let admin_if_shares_after = spot_market
+        .insurance_fund
+        .total_shares
+        .safe_sub(user_if_shares_before)?;
+
+    emit!(InsuranceFundStakeRecord {
+        ts: now,
+        user_authority: admin_pubkey,
```

Comment:
does it really make sense to associate it w admin key? its really the protocol shares? maybe use drift signer